ecl/hqlcpp/hqlhtcpp.cpp

```diff
@@ -18454,6 +18454,17 @@ IHqlExpression * HqlCppTranslator::doBuildRegexCompileInstance(BuildCtx & ctx, I
                 match = declareCtx->queryMatchExpr(searchKey);
                 if (match)
                     return match->queryExpr();
+
+                //Most regexes will have been checked in the parser, but not if it is the result of a constant fold.
+                //Check again because an error thrown in a static constructor of a dynamically loaded dll can cause
+                //the process to abort HPCC-30735
+                Owned<IException> e = checkRegexSyntax(pattern);
```

Comment:
If it is checked here, is it necessary to check it in the parser too?

Reply:
Yes, because in the parser it will report an error that directly links to the line with the problem.  Here it is unlikely to have any indication where the problem is.